pkg/daemon/ceph/osd/remove.go

```diff
@@ -58,8 +58,29 @@ func RemoveOSDs(context *clusterd.Context, clusterInfo *client.ClusterInfo, osds
 		if status == upStatus {
 			logger.Infof("osd.%d is healthy. It cannot be removed unless it is 'down'", osdID)
 			continue
+		} else {
+			logger.Infof("osd.%d is marked 'DOWN'", osdID)
+		}
+
+		// Check we can remove the OSD
+		// Loop forever until the osd is safe-to-destroy
+		if !forceOSDRemoval {
+			for {
+				isSafeToDestroy, err := client.OsdSafeToDestroy(context, clusterInfo, osdID)
+				if err != nil {
+					logger.Errorf("failed to check if osd %d is ok to stop or not ok to stop, retrying in 1m. %v", osdID, err)
+					time.Sleep(1 * time.Minute)
+				}
+				if isSafeToDestroy {
+					logger.Infof("osd.%d is ok to destroy", osdID)
+					break
+				}
+			}
+		} else {
+			logger.Infof("osd.%d might NOT be ok to destroy but force removal is enabled", osdID)
```

Comment:
How about if we check if the osd is safe to destroy so we can log that status even though we are forcing it.

Reply:
I was hesitant but can do.